lua/lspconfig/util.lua

```diff
@@ -373,4 +373,24 @@ function M.get_other_matching_providers(filetype)
   return other_matching_configs
 end
 
+function M.write_async(path, txt)
+  uv.fs_open(path, "w", 438, function(open_err, fd)
+    assert(not open_err, open_err)
+    uv.fs_write(fd, txt, -1, function(write_err)
+      assert(not write_err, write_err)
+      uv.fs_close(fd, function(close_err)
+        assert(not close_err, close_err)
+      end)
+    end)
+  end)
+end
+
+function M.read_async(path)
+  local fd = assert(vim.loop.fs_open(path, "r", 438))
+  local stat = assert(vim.loop.fs_fstat(fd))
+  local data = assert(vim.loop.fs_read(fd, stat.size, 0))
+  assert(vim.loop.fs_close(fd))
+  return data
+end
```

Comment:
this is not async.

Reply:
> I still do not understand the point of this PR. We already have method/doc generation pipeline built around explicitly annotating the configurations with validated package.json's. I don't want to add another source of truth, and I don't understand what you are trying to accomplish.

I should've done another example with an already configured server. The idea is to replace the `package_json` with the `url` entry. The file with Lua table `servers.lua` is basically a cache file that isn't going to leave this PR.